asQ/allatonce/jacobian.py

```diff
@@ -143,9 +148,9 @@ def mult(self, mat, X, Y):
         self.x.assign(X, update_halos=True, blocking=True)
 
         # assembly stage
-        fd.assemble(self.action, tensor=self.F.function)
+        fd.assemble(self.action, tensor=self.F.function, mat_type=self.mat_type)
```

Comment:
can you check if this does anything?
when assembling the 1-form, I do not think a matrix is ever assembled.